### PR TITLE
test: Move run-all-gcpc-modules to GitHub Actions 

### PR DESCRIPTION
**Pull request by aman23bedi**
**Description of your changes:**
Moved `run-all-gcpc-modules` test to github actions. 

Tested the workflows on personal repo [PR](https://github.com/aman23bedi/pipelines/pull/1)
Resolves: #11154

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by aman23bedi** [comment]
@hbelmiro

**Review by hbelmiro — APPROVED** [review]
/lgtm

Thank you for working on this @aman23bedi.

Don't forget to remove the existing Prow configuration from https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml
Once removed, please rebase this PR so the Prow test doesn't block the merge.

File: .github/workflows/gcpc-modules-tests.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  all-gcpc-tests:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
```
Comment (diegolovison) [inline .github/workflows/gcpc-modules-tests.yml:22]: Move to python 3.9

**Review by diegolovison — CHANGES_REQUESTED** [review]

**Comment by hbelmiro** [comment]
@aman23bedi can you please add a link to https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2362 in the [description of this PR](https://github.com/kubeflow/pipelines/pull/11157#issue-2498448958) so we can quickly find it?
You can use https://github.com/kubeflow/pipelines/pull/10926#issue-2360648987 as a reference.

**Comment by hbelmiro** [comment]
`test-run-all-gcpc-modules` is expected to fail because the script it invokes was removed.

**Comment by hbelmiro** [comment]
@aman23bedi https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2362 was merged.
Can you please squash your commits and rebase?

cc @DharmitD

**Comment by chensun** [comment]
> @aman23bedi [GoogleCloudPlatform/oss-test-infra#2362](https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2362) was merged. Can you please squash your commits and rebase?
> 
> cc @DharmitD

I can simply merge this manually.

**Pull request merged**